model/message.go

```diff
@@ -29,6 +31,27 @@ type Message struct {
 	Headers   http.Header
 	AgeMillis *int
 	QueueName string
+	// TODO: Do we want to add all these keys to Message? Or should they be
+	// parsed as individual strings and added to the span in
+	// processor/otel/consumer.go?
+	RoutingKey string
+	System     string
+	Operation  string
+}
+
+// Name returns a Span name for a message following the transaction/span naming schema.
+// https://github.com/elastic/apm/blob/master/specs/agents/tracing-instrumentation-messaging.md#naming
+func (m *Message) Name() string {
+	return fmt.Sprintf("%s %s %s %s",
+		m.System, strings.ToUpper(m.Operation), m.direction(), m.QueueName,
+	)
+}
+
+func (m *Message) direction() string {
+	if m.Operation == "send" {
+		return "to"
+	}
+	return "from"
```

Comment:
Can we keep this logic in `processor/otel` until we need to share it with something else?